# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/nsDocShell.cpp
# Commit: 64e118a38c1d
# Full Hash: 64e118a38c1d4ae364b6ddfc618909546c1b92f2
# Author: Dan Glastonbury <dan.glastonbury@gmail.com>
# Date: 2020-07-21 09:42:41
# Regressor Bug: 1646899
# File Overlap Count: 1
# Description:
#   Bug 1646899 - P3: Modify CanUseDocumentChannel to take necessary vars. r=mattwoodrow
#   
#   Allows checking if DocumentChannel can be used with out creating an
#   nsDocShellLoadState object.
#   
# ==============================================================================

diff -r fc96ea4de326 -r 64e118a38c1d docshell/base/nsDocShell.cpp
--- a/docshell/base/nsDocShell.cpp	Tue Jul 21 01:01:00 2020 +0000
+++ b/docshell/base/nsDocShell.cpp	Tue Jul 21 01:01:03 2020 +0000
@@ -8857,7 +8857,8 @@
   // Similar check will be performed by the ParentProcessDocumentChannel if in
   // use.
   if (XRE_IsE10sParentProcess() &&
-      !DocumentChannel::CanUseDocumentChannel(aLoadState) &&
+      !DocumentChannel::CanUseDocumentChannel(aLoadState->URI(),
+                                              aLoadState->LoadFlags()) &&
       !CanLoadInParentProcess(aLoadState->URI())) {
     return NS_ERROR_FAILURE;
   }
@@ -9748,7 +9749,8 @@
       mBrowsingContext, Some(uriModified), Some(isXFOError));
 
   nsCOMPtr<nsIChannel> channel;
-  if (DocumentChannel::CanUseDocumentChannel(aLoadState)) {
+  if (DocumentChannel::CanUseDocumentChannel(aLoadState->URI(),
+                                             aLoadState->LoadFlags())) {
     channel = DocumentChannel::CreateDocumentChannel(aLoadState, loadInfo,
                                                      loadFlags, this, cacheKey,
                                                      uriModified, isXFOError);